crypto/src/main/kotlin/web5/sdk/crypto/AwsKeyManager.kt

```diff
@@ -0,0 +1,223 @@
+    val keySpec = getAlgorithmDetails(algorithm).keySpec
+    val createKeyRequest = CreateKeyRequest()
+      .withKeySpec(keySpec)
+      .withKeyUsage(KeyUsageType.SIGN_VERIFY)
+    val createKeyResponse = kmsClient.createKey(createKeyRequest)
+    val keyId = createKeyResponse.keyMetadata.keyId
+
+    val publicKey = getPublicKey(keyId)
+    val alias = getDefaultAlias(publicKey)
+    setKeyAlias(keyId, alias)
+    return alias
+  }
+
+  /**
+   * Retrieves the public key associated with a previously stored private key, identified by the provided alias.
+   *
+   * @param keyAlias The alias referencing the stored private key.
+   * @return The associated public key in JWK (JSON Web Key) format.
+   * @throws [AWSKMSException] for any error originating from the [AWSKMS] client
+   */
+  override fun getPublicKey(keyAlias: String): JWK {
+    val getPublicKeyRequest = GetPublicKeyRequest().withKeyId(keyAlias)
+    val publicKeyResponse = kmsClient.getPublicKey(getPublicKeyRequest)
+    val publicKey = convertToJavaPublicKey(publicKeyResponse.publicKey)
+
+    val algorithmDetails = getAlgorithmDetails(publicKeyResponse.keySpec.enum())
+    val jwkBuilder = when (publicKey) {
+      is ECPublicKey -> ECKey.Builder(algorithmDetails.curve, publicKey)
+      else -> throw IllegalArgumentException("Unknown key type $publicKey")
+    }
+    return jwkBuilder
+      .algorithm(algorithmDetails.algorithm)
+      .keyID(keyAlias)
+      .keyUse(KeyUse.SIGNATURE)
+      .build()
+  }
+
+  /**
+   * Signs the provided payload using the private key identified by the provided alias.
+   *
+   * @param keyAlias The alias referencing the stored private key.
+   * @param signingInput The data to be signed.
+   * @return The signature in JWS R+S format
+   * @throws [AWSKMSException] for any error originating from the [AWSKMS] client
+   */
+  override fun sign(keyAlias: String, signingInput: ByteArray): ByteArray {
+    val keySpec = fetchKeySpec(keyAlias)
+    val algorithmDetails = getAlgorithmDetails(keySpec)
+    //Pre-hash the input because AWS limits the message to 4096 bytes
+    val hashedMessage = shaDigest(algorithmDetails, signingInput)
+
+    val signRequest = SignRequest()
+      .withKeyId(keyAlias)
+      .withMessageType(MessageType.DIGEST)
+      .withMessage(hashedMessage.asByteBuffer())
+      .withSigningAlgorithm(algorithmDetails.signingAlgorithm)
+    val signResponse = kmsClient.sign(signRequest)
+    val derSignatureBytes = signResponse.signature.array()
+    return transcodeDerSignatureToConcat(derSignatureBytes, algorithmDetails.algorithm)
+  }
+
+  /**
+   * Return the alias of [publicKey], as was originally returned by [generatePrivateKey].
+   *
+   * @param publicKey A public key in JWK (JSON Web Key) format
+   * @return The alias belonging to [publicKey]
+   */
+  override fun getDefaultAlias(publicKey: JWK): String {
+    val jwkThumbprint = publicKey.computeThumbprint()
```

Comment:
Curious why thumbprint instead of using the `kid` field?

Reply:
This does go into `kid` up in `generatePrivateKey`

Reply:
@andresuribe87 because `kid` isn't a required property of JWKs ([reference](https://datatracker.ietf.org/doc/html/rfc7517#section-4.5)). 

IMO, Using [JWK thumbprints](https://datatracker.ietf.org/doc/html/rfc7638) makes it so there's a deterministic mapping between a DID's verification method (presuming that the verification method has `publicKeyJwk`, a requirement we intend to surface as part of an upcoming interop profile) and a key manager's key alias without having to rely on a [Verification Method ID](https://www.w3.org/TR/did-core/#verification-methods) or [JWK kid](https://datatracker.ietf.org/doc/html/rfc7517#section-4.5). Especially because a verification method id and a kid (even if both are present) don't have to match